internal/operands/template-validator/reconcile.go

```diff
@@ -173,15 +172,113 @@ func reconcileDeployment(request *common.Request) (common.ReconcileResult, error
 		Reconcile()
 }
 
-func addPlacementFields(deployment *apps.Deployment, nodePlacement *lifecycleapi.NodePlacement) {
-	if nodePlacement == nil {
+// Merge all Tolerations, Affinity and NodeSelectors from NodePlacement into pod spec
+func injectPlacementMetadata(podSpec *v1.PodSpec, componentConfig *v1beta1.TemplateValidator) {
+	if componentConfig == nil || componentConfig.Placement == nil {
 		return
 	}
+	if podSpec == nil {
+		podSpec = &v1.PodSpec{}
+	}
+	nodePlacement := componentConfig.Placement
+	if len(nodePlacement.NodeSelector) == 0 {
```

Comment:
Any reason why this is being done instead of `if nodePlacement.NodeSelector == nil`? If `nodePlacement.NodeSelector` is an empty map, can't we just continue and avoid re-allocating a new map?